app/uk/gov/hmrc/calculatenifrontend/controllers/TablesController.scala

```diff
@@ -55,12 +63,102 @@ class TablesController @Inject()(
 
     val selectedCategory = category.filter(categories.contains).getOrElse(categories.head)
 
-    val filteredBand: List[(String, RateDefinition)] =
-      band.filter{
-        case (_, v) => (v.employee.keySet ++ v.employer.keySet).contains(selectedCategory)
-      }.toList.sortBy(_._2.year.lowerValue.get)
-
+    val filteredBand: List[(String, RateDefinition)] = band.filter{
+      case (_, v) => (v.employee.keySet ++ v.employer.keySet).contains(selectedCategory)
+    }.toList.sortBy(_._2.year.lowerValue.get)
 
     Ok(classOnePage(selectedInterval, intervals, selectedCategory, categories, filteredBand))
   }
+
+  def classTwo(
+    date: Option[LocalDate]
+  ): Action[AnyContent] = Action { implicit request =>
+    val intervals: List[(LocalDate, String)] = intervalDropdown(ni.classTwo)
+    val dateP = date.getOrElse(intervals.head._1)
+
+    ni.classTwo.at(dateP) match {
+      case Some(data) =>
+        val selectedInterval = ni.classTwo.keySet.find(_.contains(dateP)).get
+        val lowerBound = selectedInterval.lowerValue.get
+        
+        val noOfWeeks = selectedInterval.numberOfWeeks.get
+        val response = List (
+          "Term Date" -> LocalDate.of(lowerBound.getYear, 4, 9), // unknown... but always the 9th of april
```

Comment:
how sure are you that this is and always will be true?

Reply:
Not massively. I'm going off of my notes from a conversation with Andrew A. The plan is to demo this to him and ask for confirmation.